File: src/main/java/com/example/DemoGraphQL/resolver/Query.java
package com.example.DemoGraphQL.resolver;

import com.coxautodev.graphql.tools.GraphQLQueryResolver;
import com.example.DemoGraphQL.model.Author;
import com.example.DemoGraphQL.model.Book;
import com.example.DemoGraphQL.repository.AuthorRepository;
import com.example.DemoGraphQL.repository.BookRepository;
import org.springframework.data.domain.PageRequest;

public class Query implements GraphQLQueryResolver {
    private BookRepository bookRepository;
    private AuthorRepository authorRepository;

    public Query(AuthorRepository authorRepository, BookRepository bookRepository) {
        this.authorRepository = authorRepository;
        this.bookRepository = bookRepository;
    }

    public Iterable<Book> findAllBooks() { return bookRepository.findAll(PageRequest.of(0, 2)); }

    public Iterable<Author> findAllAuthors() {
        return authorRepository.findAll(PageRequest.of(1, 2));
    }

    public long countBooks() {
        return bookRepository.count();
    }
    public long countAuthors() {
        return authorRepository.count();
    }
}

File: src/main/java/com/example/DemoGraphQL/resolver/Query.kt
package com.example.DemoGraphQL.resolver

import com.coxautodev.graphql.tools.GraphQLQueryResolver
import com.example.DemoGraphQL.model.Author
import com.example.DemoGraphQL.model.Book
import com.example.DemoGraphQL.repository.AuthorRepository
import com.example.DemoGraphQL.repository.BookRepository
import org.springframework.data.domain.PageRequest

class Query(private val authorRepository: AuthorRepository, private val bookRepository: BookRepository) : GraphQLQueryResolver {

    fun findAllBooks(): Iterable<Book> {
        return bookRepository.findAll(PageRequest.of(0, 2))
    }

    fun findAllAuthors(): Iterable<Author> {
        return authorRepository.findAll(PageRequest.of(1, 2))
    }

    fun countBooks(): Long {
        return bookRepository.count()
    }

    fun countAuthors(): Long {
        return authorRepository.count()
    }
}
